static/app/views/insights/agents/components/drawer.tsx

```diff
@@ -40,14 +39,19 @@ const TraceViewDrawer = memo(function TraceViewDrawer({
 }) {
   const organization = useOrganization();
   const {nodes, isLoading, error} = useAITrace(traceSlug);
-  const [selectedNodeKey, setSelectedNodeKey, removeSelectedNodeParam] =
-    useLocationSyncedState(DrawerUrlParams.SELECTED_SPAN, decodeScalar);
+  const [selectedNodeKey, setSelectedNodeKey] = useQueryState(
+    DrawerUrlParams.SELECTED_SPAN,
+    parseAsString.withOptions({history: 'replace'})
+  );
+
+  const clearSelectedNodeKey = useEffectEvent(() => {
+    void setSelectedNodeKey(null);
+  });
 
   useEffect(() => {
     return () => {
-      removeSelectedNodeParam();
+      clearSelectedNodeKey();
     };
-    // eslint-disable-next-line react-hooks/exhaustive-deps -- only run on mount
   }, []);
```

Comment:
I think this could still be just a regular effect with `setSelectedNodeKey` in the dependency array because it’s referentially stable.

Also, just for my understanding: why do we need to clear the url param when this component mounts? Shouldn’t we remove the param from the link that navigates here if we want that? Otherwise, there will still be one render cycle where the value is available in state ...